### PR TITLE
PMM-6640 API for HAProxy.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

**Review by askomorokhov — APPROVED** [review]

**Review by BupycHuk — CHANGES_REQUESTED** [review]

**Review by BupycHuk — CHANGES_REQUESTED** [review]
let's add `ha` here https://github.com/percona/pmm/blob/release%2F2.15/api/Makefile#L34

File: api/managementpb/haproxy.proto [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import "managementpb/metrics.proto";
+import "managementpb/service.proto";
+
+// Add HAProxy
+
+message AddHAProxyRequest {
+  // Node identifier on which an external exporter is been running.
+  // runs_on_node_id always should be passed with node_id.
+  // Exactly one of these parameters should be present: node_id, node_name, add_node.
+  string runs_on_node_id = 1;
```
Comment (BupycHuk) [inline api/managementpb/haproxy.proto:20]: let's rename `runs_on_node_id` to `node_id` and remove duplicated `node_id` below

Comment (BupycHuk) [reply api/managementpb/haproxy.proto:20]: and update comments

File: api/managementpb/haproxy.proto [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  // Environment name.
+  string environment = 9;
+  // Cluster name.
+  string cluster = 10;
+  // Replication set name.
+  string replication_set = 11;
+  // Custom user-assigned labels for Service.
+  map<string, string> custom_labels = 15;
+  // Group name of HAProxy service.
+  string group = 16;
```
[the remark this answers is not in the document]
Comment (JiriCtvrtka) [reply api/managementpb/haproxy.proto]: Removed.

File: api/managementpb/haproxy.proto (hunk 1 above)
Comment (JiriCtvrtka) [reply api/managementpb/haproxy.proto]: Removed and updated.

File: api/inventorypb/services.proto [hunk 3]
```diff
@@ -145,6 +146,26 @@ message ExternalService {
...
+  // Environment name.
+  string environment = 4;
+  // Cluster name.
+  string cluster = 5;
+  // Replication set name.
+  string replication_set = 6;
+  // Custom user-assigned labels.
+  map<string, string> custom_labels = 7;
+  // Group name of HAProxy service.
+  string group = 8;
```
Comment (BupycHuk) [inline api/inventorypb/services.proto:166]: let's remove this group too

File: api/inventorypb/services.proto [hunk 4]
```diff
@@ -367,6 +390,37 @@ message AddExternalServiceResponse {
...
+  // Environment name.
+  string environment = 3;
+  // Cluster name.
+  string cluster = 4;
+  // Replication set name.
+  string replication_set = 5;
+  // Custom user-assigned labels.
+  map<string, string> custom_labels = 6;
+  // Group name of HAProxy service.
+  string group = 7;
```
Comment (BupycHuk) [inline api/inventorypb/services.proto:417]: and here too.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: api/inventorypb/services.proto (hunk 3 above)
Comment (JiriCtvrtka) [reply api/inventorypb/services.proto]: Removed.

File: api/inventorypb/services.proto (hunk 4 above)
Comment (JiriCtvrtka) [reply api/inventorypb/services.proto]: Removed.

File: api/managementpb/haproxy.proto [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+message AddHAProxyResponse {
+  inventory.HAProxyService service = 1;
+  inventory.ExternalExporter external_exporter = 2;
+}
+
+// HAProxy service provides public Management API methods for HAProxy Service.
+service HAProxy {
+  // AddHAProxy adds HAProxy service and adds external exporter.
+  // It automatically adds a service to inventory, which is running on provided "node_id",
+  // then adds an "external exporter" agent to inventory, which is running on provided "runs_on_node_id".
```
Comment (BupycHuk) [inline api/managementpb/haproxy.proto:76]: please update the comment

**Review by BupycHuk — APPROVED** [review]
please update a comment for AddHAProxy, everything else looks good to me

**Pull request merged**